lib/modules/datasource/rubygems/versions-datasource.ts

```diff
@@ -1,38 +1,69 @@
 import { logger } from '../../../logger';
 import { ExternalHostError } from '../../../types/errors/external-host-error';
 import { getElapsedMinutes } from '../../../util/date';
+import { HttpError } from '../../../util/http';
 import { newlineRegex } from '../../../util/regex';
 import { Datasource } from '../datasource';
 import type { GetReleasesConfig, ReleaseResult } from '../types';
 
-let lastSync = new Date('2000-01-01');
-let packageReleases: Record<string, string[]> = Object.create(null); // Because we might need a "constructor" key
-let contentLength = 0;
+type RegistryCache = {
+  lastSync: Date;
+  packageReleases: Record<string, string[]>; // Because we might need a "constructor" key
+  contentLength: number;
+  isSupported: boolean;
+};
+
+const registryCaches: { [key: string]: RegistryCache } = {};
 
 // Note: use only for tests
 export function resetCache(): void {
-  lastSync = new Date('2000-01-01');
-  packageReleases = Object.create(null);
-  contentLength = 0;
+  Object.keys(registryCaches).forEach((key) => {
+    registryCaches[key].lastSync = new Date('2000-01-01');
+    registryCaches[key].packageReleases = Object.create(null);
+    registryCaches[key].contentLength = 0;
+    registryCaches[key].isSupported = false;
+  });
 }
 
-export class RubyGemsOrgDatasource extends Datasource {
-  constructor(override readonly id: string) {
+export class VersionsDatasource extends Datasource {
+  private registryUrl: string;
+  private registryCache: RegistryCache;
+
+  constructor(override readonly id: string, registryUrl: string) {
     super(id);
+    this.registryUrl = registryUrl;
+    if (!registryCaches[registryUrl]) {
+      registryCaches[registryUrl] = {
+        lastSync: new Date('2000-01-01'),
+        packageReleases: Object.create(null),
+        contentLength: 0,
+        isSupported: false,
```

Comment:
Defaults to false?

Reply:
I think my thought was, assume it's not supported until you check (especially if it's considered an "old" standard)... but I could go either way. What would be the pro/con of `true` vs `false` here from your perspective?